hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -219,10 +220,22 @@ protected HoodieMetadataPayload(String key, int type,
                                                                                Option<Map<String, Long>> filesAdded,
                                                                                Option<List<String>> filesDeleted) {
     Map<String, HoodieMetadataFileInfo> fileInfo = new HashMap<>();
-    filesAdded.ifPresent(
-        m -> m.forEach((filename, size) -> fileInfo.put(filename, new HoodieMetadataFileInfo(size, false))));
-    filesDeleted.ifPresent(
-        m -> m.forEach(filename -> fileInfo.put(filename, new HoodieMetadataFileInfo(0L, true))));
+    filesAdded.ifPresent(filesMap ->
+        fileInfo.putAll(
+            filesMap.entrySet().stream().collect(
```

Comment:
minor: stream().collect() is going to double the memory used here as it creates a new map first before adding to fileInfo. I feel forEach is better suited here.

Not a big issue as these are just filenames.

Reply:
I'd argue that this short-lived doubling is actually less impactful on GC than log(N) (where N is the number of filenames) resizes of the HashMap which is happening with individual `put`s